src/core/shared_memory_manager.h

```diff
@@ -32,8 +32,21 @@
 #include "src/core/server_status.pb.h"
 #include "src/core/status.h"
 
+#ifdef TRTIS_ENABLE_GPU
+#include <cuda_runtime_api.h>
+#endif  // TRTIS_ENABLE_GPU
+
 namespace nvidia { namespace inferenceserver {
 
+#ifdef TRTIS_ENABLE_GPU
+struct cudaIpcHandle {
```

Comment:
This still doesn't follow our naming conventions and it should be nested inside SharedMemoryManager. Also, how is the user/client going to know they must generate this structure? We should really just be sending cudaIpcMemHandle_t... and I don't think we even need the complicated way of sending the handle. We should discuss in depth before you proceed.

Reply:
Restructured the CudaRegisterSharedMemory to use only the handle and an entirely different structure.